src/MerchantCenter/AccountService.php

```diff
@@ -330,6 +333,23 @@ private function setup_account_steps() {
 							$merchant->claimwebsite();
 						}
 						break;
+					case 'link_ads':
+						$ads_id = $this->options->get_ads_id();
+						// As MC and Ads account can be connected interchangeably, this check is necessary.
+						if ( ! empty( $ads_id ) && ! empty( $merchant_id ) ) {
+							$this->link_ads_account();
+						} else {
+							/**
+							 * Mark the step as pending, save the state and
+							 * continue the foreach loop with `continue 2`.
+							 *
+							 * Marking the step as pending will make sure to run this step next time.
+							 */
+							$state[ $name ]['status'] = MerchantAccountState::STEP_PENDING;
+							$this->state->update( $state );
+							continue 2;
+						}
+						break;
```

Comment:
Same as the previous suggestion, we can simplify this code to make it flow better and still have it return an error if the merchant account isn't set:

```suggestion
					case 'link_ads':
						// Continue to next step if Ads account is not connected yet.
						if ( ! $this->options->get_ads_id() ) {
							// Save step as pending and continue the foreach loop with `continue 2`.
							$state[ $name ]['status'] = MerchantAccountState::STEP_PENDING;
							$this->state->update( $state );
							continue 2;
						}

						$this->link_ads_account();
						break;
```